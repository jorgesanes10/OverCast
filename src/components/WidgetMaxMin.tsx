import { Grid2 } from '@mui/material';
import { ReactNode } from 'react';
import styled from 'styled-components';
import { Thermostat } from '@mui/icons-material';

interface WidgetMaxMinProps {
  maxTemperature: ReactNode;
  minTemperature: ReactNode;
  ariaLabel: string;
}

export default function WidgetMaxMin({
  maxTemperature,
  minTemperature,
  ariaLabel,
}: WidgetMaxMinProps) {
  return (
    <StyledWidgetMaxMin aria-label={ariaLabel} container>
      <Grid2 aria-hidden>
        <p>
          <Thermostat /> Min
        </p>
        <p className="value">{minTemperature}</p>
      </Grid2>
      <Grid2 aria-hidden>
        <p>
          <Thermostat /> Max
        </p>
        <p className="value">{maxTemperature}</p>
      </Grid2>
    </StyledWidgetMaxMin>
  );
}

const StyledWidgetMaxMin = styled(Grid2)`
  padding: 12px;
  color: #fff;
  background-color: #ffffff30;
  border-radius: 8px;

  display: flex;
  justify-content: space-evenly;

  p {
    margin: 0;
    font-weight: 100;
  }

  p.value {
    margin: 24px 0;
    font-size: 28px;
    align-self: center;
    font-weight: 400;
  }

  svg {
    position: relative;
    top: 5px;
  }
`;
