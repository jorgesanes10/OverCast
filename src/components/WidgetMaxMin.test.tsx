import { render, screen } from '@testing-library/react';
import WidgetMaxMin from './WidgetMaxMin.tsx';

describe('WidgetmaxMin', () => {
  test('renders the component', () => {
    const { container } = render(
      <WidgetMaxMin
        maxTemperature="30 F"
        minTemperature="20 F"
        ariaLabel="My aria label"
      />,
    );

    expect(screen.getByText('30 F')).toBeDefined();
    expect(screen.getByText('20 F')).toBeDefined();

    expect(container).toMatchSnapshot();
  });
});
