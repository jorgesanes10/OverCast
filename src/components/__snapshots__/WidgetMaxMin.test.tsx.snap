// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetmaxMin renders the component 1`] = `
<div>
  <div
    aria-label="My aria label"
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row sc-KXCwU byIAil widget css-1jxwjqx-MuiGrid2-root"
  >
    <div
      aria-hidden="true"
      class="MuiGrid2-root MuiGrid2-direction-xs-row css-1fzlhpv-MuiGrid2-root"
    >
      <p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ThermostatIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 13V5c0-1.66-1.34-3-3-3S9 3.34 9 5v8c-1.21.91-2 2.37-2 4 0 2.76 2.24 5 5 5s5-2.24 5-5c0-1.63-.79-3.09-2-4m-4-2V5c0-.55.45-1 1-1s1 .45 1 1v1h-1v1h1v2h-1v1h1v1z"
          />
        </svg>
         Min
      </p>
      <p
        class="value"
      >
        20 F
      </p>
    </div>
    <div
      aria-hidden="true"
      class="MuiGrid2-root MuiGrid2-direction-xs-row css-1fzlhpv-MuiGrid2-root"
    >
      <p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="ThermostatIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 13V5c0-1.66-1.34-3-3-3S9 3.34 9 5v8c-1.21.91-2 2.37-2 4 0 2.76 2.24 5 5 5s5-2.24 5-5c0-1.63-.79-3.09-2-4m-4-2V5c0-.55.45-1 1-1s1 .45 1 1v1h-1v1h1v2h-1v1h1v1z"
          />
        </svg>
         Max
      </p>
      <p
        class="value"
      >
        30 F
      </p>
    </div>
  </div>
</div>
`;
